GolfExample_TCAPI/index.html

```diff
@@ -41,12 +39,75 @@
                 }
             );
 
-        tincan.sendStatement(
-            {
-                verb: "attempted",
-                context: GolfExample.getContext()
+        var bookmarkingTracker = function(){
+            this.currentPage = 0;
+            this.startTimeStamp = new Date();
+            this.startAttemptDuration = 0;
+            this.attemptComplete = false;
+        };
+
+        bookmarkingTracker.prototype = {
+            initFromBookmark: function(bookmark){
+                this.setPage(parseInt(bookmark.location, 10));
+                this.setStartDuration(bookmark.attemptDuration);
+                this.getCompletion(bookmark.attemptComplete);
+            },
+            reset: function(){
+                this.setPage(0);
+                this.setStartDuration(0);
+                this.setCompletion(false);
+            },
+            save: function (callback){
+                var bookmarking = {
+                    location: this.currentPage,
+                    attemptDuration: this.getAttemptDuration(),
+                    attemptComplete: this.attemptComplete
+                };
+                tincan.setState("bookmarking-data", bookmarking, {
+                    contentType: "application/json",
+                    callback: callback
+                });
+            },
+            get: function(){
+                var stateResult = tincan.getState("bookmarking-data");
+                if (stateResult.err === null && stateResult.state !== null && stateResult.state.contents !== "") {
+                    return stateResult.state.contents;
+                } else {
```

Comment:
Don't cuddle `else`s, and in this case you don't need the `else` because the only way out of the above `if` is through `return`.